prymer/offtarget/bwa.py

```diff
@@ -334,13 +330,17 @@ def map_all(self, queries: list[Query]) -> list[BwaResult]:
         for query in queries:
             fastq_str = query.to_fastq(reverse_complement=self.reverse_complement)
             self._subprocess.stdin.write(fastq_str)
-        self.__signal_bwa()  # forces the input to be sent to the underlying process.
+
+        # Force the input to be sent to the underlying process.
+        self.__signal_bwa()
 
         # Read back the results
         results: list[BwaResult] = []
         for query in queries:
             # get the next alignment and convert to a result
-            results.append(self._to_result(query=query, rec=next(self._reader)))
+            line: str = next(self._subprocess.stdout).strip()
```

Comment:
I wonder if to be extra safe we should add a check here that if the line starts with `@` to keep reading lines?

Reply:
I like it! This is a good use case for `assert` since you can turn off asserts when running Python via "optimization" mode.

https://docs.python.org/3/using/cmdline.html#cmdoption-O